src/tools/components/SequenceSearchLoader.tsx

```diff
@@ -0,0 +1,245 @@
+    return uniProtKBApiUrls.entry(cleanedInput);
+  }
+
+  // UniProtKB ID
+  const query = queryString.stringify({
+    query: `id:${cleanedInput}`,
+    fields:
+      'sequence,id,reviewed,protein_name,organism_name,protein_existence,sequence_version',
+  });
+
+  return `${uniProtKBApiUrls.search}?${query}`;
+};
+
+export type ParsedSequence = {
+  sequence: string;
+  raw: string;
+  header: string;
+  valid: boolean;
+  likelyType: 'na' | 'aa' | null;
+  message: string | null;
+  name?: string;
+};
+
+type NetworkResponses = { results: APISequenceData[] } | APISequenceData;
+
+const SequenceSearchLoader = forwardRef<
+  { reset: () => void },
+  { onLoad: (event: ParsedSequence[]) => void }
+>(({ onLoad }, ref) => {
+  const [accessionOrID, setAccessionOrID] = useState('');
+  const [pasteLoading, setPasteLoading] = useState(false);
+  const dispatch = useDispatch();
+
+  useImperativeHandle(ref, () => ({
+    reset: () => setAccessionOrID(''),
+  }));
+
+  // used to keep a reference to the previously generated sequence string
+  const sequenceRef = useRef('');
+
+  const urlForAccessionOrID = getURLForAccessionOrID(accessionOrID);
+  const { data, loading } = useDataApi<NetworkResponses>(urlForAccessionOrID);
+
+  let entry: APISequenceData | null = null;
+  if (data) {
+    if ('results' in data) {
+      [entry] = data.results;
+    } else {
+      entry = data;
+    }
+  }
+
+  // no entry found, probably invalid query, reset ref
+  if (!entry) {
+    sequenceRef.current = '';
+  }
+
+  useEffect(() => {
+    if (!entry) {
+      return;
+    }
+
+    if (!urlForAccessionOrID) {
+      onLoad([
+        {
+          raw: '',
+          header: '',
+          sequence: '',
+          valid: false,
+          likelyType: null,
+          message: null,
+        },
+      ]);
+      return;
+    }
+
+    const sequence = `${entryToFASTAWithHeaders(entry)}\n`;
+
+    if (sequence === sequenceRef.current) {
+      // if the new generated sequence would be the same than the previously
+      // generated one, don't do anything
+      // we must have end up here because something else update (e.g. the user
+      // manually updated the sequence)
+      return;
+    }
+
+    // set ref to the value of the sequence we are about to set
+    sequenceRef.current = sequence;
+
+    onLoad([
+      {
+        raw: sequence,
+        // no need to fill the rest, it will be parsed again later
+        header: '',
+        sequence: '',
+        valid: true,
+        likelyType: null,
+        message: null,
+        // name as a NCBI ID formatted UniProt-style
+        name: entry.uniProtkbId
+          ? `${entry.entryType === EntryType.REVIEWED ? 'sp' : 'tr'}|${
+              entry.primaryAccession
+            }|${entry.uniProtkbId}`
+          : '',
+      },
+    ]);
+  }, [entry, onLoad, urlForAccessionOrID, accessionOrID]);
+
+  const handlePaste = useCallback(
+    async (event: ClipboardEvent) => {
```

Comment:
I wasn't able to do the same logic only with onChange.
Especially, it's messing up when there's already text in the field and the user pastes. If it wasn't for that I would have used the onChange approach.
Regarding clipboard event, I've added a check to escape without failing in case it is not supported, but looking at [this](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData), even if it is officially "experimental", it is widely supported

Reply:
I think we this one it's better to have a chat within the team. My issue with using unstable features -- even widely supported, is that they might drop-off from some vendors or the API might change and start to become more and more divert between different vendors.

I haven't tried to see if I can make it work with only `onChange` but let's wait and talk with others and then if the decision was to not use clipboard events, then we can try to find another solution.

Reply:
While considered experimental, it seems like ClipboardEvent.clipboardData is has been supported for a while [going back to IE6 (2001)](https://caniuse.com/#feat=mdn-api_clipboardevent_clipboarddata). At the same time it seems like Chrome and Safari have [had some issues in 2018](https://stackoverflow.com/questions/48186920/clipboard-event). Would a [poly-fill](https://www.npmjs.com/package/clipboard-polyfill) be useful here for insurance?

Reply:
The poly-fill I linked is for loading into the clipboard 😦 not the paste event